components/cluster/command/deploy.go

```diff
@@ -77,17 +77,7 @@ func newDeploy() *cobra.Command {
 				teleTopology = string(data)
 			}
 
-			return manager.Deploy(
-				clusterName,
-				version,
-				topoFile,
-				opt,
-				postDeployHook,
-				skipConfirm,
-				gOpt.OptTimeout,
-				gOpt.SSHTimeout,
-				gOpt.SSHType,
-			)
+			return cm.Deploy(clusterName, version, topoFile, opt, postDeployHook, skipConfirm, gOpt)
```

Comment:
It's better to keep the default manager defined in `manager` package:

```
var clusterManager *clusterManager

func Deploy(...) {
    return clusterManager.Deploy(...)
}
```

Reply:
I'm sorry, but I'm curious about the advantage of this design pattern.  We used a singleton instance to handle all `Manager` related works, the former declares the instance in user-space, while the latter in the package-space. But the later one seems duplicated, do we need to declare a one-one proxy method for the public methods?